### ISSUE TITLE
[Feature Request] Allow dynamic ssl/tls on the client.

### ISSUE THREAD

**Issue by robcao** [temporalio/sdk-dotnet#141]
### Is your feature request related to a problem? Please describe.

We have the following business case:

Due to compliance and security requirements, our TLS certificates are short lived (a couple of hours). Our Temporal workers live longer than these TLS certificates.

We would like the ability to dynamically refresh the TLS options on the Temporal client instead of having to restart the Temporal worker itself if the worker has outlived the original certificate.

As I understand, the gRPC connection itself lives in unmanaged rust, so traditional .NET approaches to dynamic certificate refresh such as https://github.com/dotnet/runtime/issues/52779 won't be applicable.

### Describe the solution you'd like

Perhaps a variant of the `TlsOptions` class [here](https://github.com/temporalio/sdk-dotnet/blob/main/src/Temporalio/Client/TlsOptions.cs), that accepts a callback instead of static values.

```
public class DynamicTlsOptions
{
     public Func<TlsCallbackContext, Task<TlsOptions>> TlsOptionsCallback { get; set; }

     public TimeSpan MinimumGuaranteedCallbackFrequency { get; set; }
}
```

It would be preferable for our use case if this callback was asynchronous, but a synchronous callback would work just fine.

For our use case, we don't require any metadata or context available to us from Temporal in the callback, provided that there are guarantees about the frequency of how often the callback is invoked (we don't necessarily need this to be a configurable setting).

**Comment by cretz** [comment]
Yes, this is something we have discussed but it would be more push based than pull based. This is dependent on https://github.com/temporalio/sdk-core/issues/477.

**Comment by cretz** [comment]
This has been completed a while ago. New clients can be created with different TLS options and can be set on a worker to have successive client calls from the worker use the newer client.
